pkg/controllers/awsloadbalancercontroller/deployment.go

```diff
@@ -377,3 +390,63 @@ func indexedContainerEnv(envs []corev1.EnvVar) map[string]corev1.EnvVar {
 	}
 	return indexed
 }
+
+func hasSecurityContextChanged(current, desired *corev1.SecurityContext) bool {
+	if desired == nil {
+		return false
+	}
+
+	if current == nil {
+		return true
+	}
+
+	if desired.Capabilities != nil {
+		if current.Capabilities == nil {
+			return true
+		}
+		cmpCapabilities := cmpopts.SortSlices(func(a, b corev1.Capability) bool { return a < b })
+		if !cmp.Equal(desired.Capabilities.Add, current.Capabilities.Add, cmpCapabilities) {
+			return true
+		}
+
+		if !cmp.Equal(desired.Capabilities.Drop, current.Capabilities.Drop, cmpCapabilities) {
+			return true
+		}
+	}
+
+	if !equalBoolPtr(current.RunAsNonRoot, desired.RunAsNonRoot) {
+		return true
+	}
+
+	if !equalBoolPtr(current.Privileged, desired.Privileged) {
+		return true
+	}
+	if !equalBoolPtr(current.AllowPrivilegeEscalation, desired.AllowPrivilegeEscalation) {
+		return true
+	}
+
+	if desired.SeccompProfile != nil {
+		if current.SeccompProfile == nil {
+			return true
+		}
+		if desired.SeccompProfile.Type != "" && desired.SeccompProfile.Type != current.SeccompProfile.Type {
+			return true
+		}
+	}
+	return false
+}
+
+func equalBoolPtr(current, desired *bool) bool {
+	if desired == nil {
+		return true
+	}
+
+	if current == nil {
+		return false
+	}
+
+	if *current != *desired {
+		return false
+	}
+	return true
+}
```

Comment:
I think `cmp.Equal()` has the same behavior for pointers as well.

Reply:
It's not the same behavior. Because this function is returns true when desired is nil but not when current is nil when desired is not. `cmp.Equal()` won't give the same behavior. The reason for the special type of comparison is discussed [here](https://github.com/openshift/aws-load-balancer-operator/pull/64#discussion_r874169605).